cnd/src/swap_protocols/rfc003/ethereum/htlc_events.rs

```diff
@@ -72,12 +83,7 @@ impl HtlcEvents<Ethereum, EtherQuantity> for Arc<dyn QueryEthereum + Send + Sync
         htlc_deployment: &Deployed<Ethereum>,
         htlc_funding: &Funded<Ethereum, EtherQuantity>,
     ) -> Box<RedeemedOrRefundedFuture<Ethereum>> {
-        htlc_redeemed_or_refunded(
-            Arc::clone(&self),
-            htlc_params,
-            htlc_deployment,
-            htlc_funding,
-        )
+        htlc_redeemed_or_refunded(self.clone(), htlc_params, htlc_deployment, htlc_funding)
```

Comment:
Is this supposed to be recursive?

Reply:
It is not recursive, it delegates to a free-function further down.

Reply:
Forgive me if I'm wrong but this seems like a function that has a single call that calls itself, am I missing something?
```

        fn htlc_redeemed_or_refunded(
            &self,
            htlc_params: HtlcParams<Ethereum, Erc20Token>,
            htlc_deployment: &Deployed<Ethereum>,
            htlc_funding: &Funded<Ethereum, Erc20Token>,
        ) -> Box<RedeemedOrRefundedFuture<Ethereum>> {
            htlc_redeemed_or_refunded(self.clone(), htlc_params, htlc_deployment, htlc_funding)
        }

```

Reply:
What is a 'free-function'?

Reply:
> What is a 'free-function'?

One that is not associated with a struct/trait :)
For example: https://stackoverflow.com/questions/4861914/what-is-the-meaning-of-the-term-free-function-in-c

The first fn is part of a trait and the other one is defined next to it because it takes a generic argument and is reused because the "ether" and "erc20" implementations of the `HtlcEvents` trait.